routes/favicon.coffee

```diff
@@ -0,0 +1,28 @@
+request = require 'request'
+fs = require 'fs'
+cachefile = 'favicon.ico'
+
+favicon = (req, res, next) ->
+	try
+		res.send fs.readFileSync cachefile
+	catch error
+		request "https://vine.co/", (error, response, body) ->
```

Comment:
> Using only async functions where possible actually was my initial approach, but then I was deterred by the deep indentation.

There are tricks to avoid this. I'm going to comment on the new patch with some suggestions.